components/dropdown-menu/index.js

```diff
@@ -60,62 +61,62 @@ class DropdownMenu extends wp.element.Component {
 	}
 
 	findActiveIndex() {
-		if ( this.menuRef ) {
-			const menu = findDOMNode( this.menuRef );
+		if ( this.menuNode ) {
 			const menuItem = document.activeElement;
-			if ( menuItem.parentNode === menu ) {
-				return findIndex( menu.children, ( child ) => child === menuItem );
+			if ( menuItem.parentNode === this.menuNode ) {
+				return findIndex( this.menuNode.children, ( child ) => child === menuItem );
 			}
 			return -1;
 		}
 	}
 
 	focusIndex( index ) {
-		if ( this.menuRef ) {
-			const menu = findDOMNode( this.menuRef );
-			if ( index < 0 ) {
-				menu.previousElementSibling.focus();
-			} else {
-				menu.children[ index ].focus();
-			}
+		if ( this.menuNode ) {
+			this.menuNode.children[ index ].focus();
 		}
 	}
 
 	focusPrevious() {
 		const i = this.findActiveIndex();
-		const prevI = i <= -1 ? -1 : i - 1;
+		const maxI = this.props.controls.length - 1;
+		const prevI = i <= 0 ? maxI : i - 1;
 		this.focusIndex( prevI );
 	}
 
 	focusNext() {
 		const i = this.findActiveIndex();
 		const maxI = this.props.controls.length - 1;
-		const nextI = i >= maxI ? maxI : i + 1;
+		const nextI = i >= maxI ? 0 : i + 1;
 		this.focusIndex( nextI );
 	}
 
+	handleKeyUp( event ) {
+		// TODO: find a better way to isolate events on nested components see GH issue #1973.
+		/*
+		 * VisualEditorBlock uses a keyup event to deselect the block. When the
+		 * menu is open we need to stop propagation after Escape has been pressed
+		 * so we use a keyup event instead of keydown, otherwise the whole block
+		 * toolbar will disappear.
+		 */
+		if ( event.keyCode === ESCAPE && this.state.open ) {
+			event.preventDefault();
+			event.stopPropagation();
+			const node = findDOMNode( this );
+			const toggle = node.querySelector( '.components-dropdown-menu__toggle' );
```

Comment:
Actually, for that matter, we could bind the `ref` to the toggle to avoid the `querySelector` as well.